domain/schema/model/sql/0034-cross-model-relation.sql

```diff
@@ -68,35 +71,46 @@ CREATE TABLE application_remote_consumer (
     CONSTRAINT fk_life_id
     FOREIGN KEY (life_id)
     REFERENCES life (id),
+    CONSTRAINT fk_application_uuid
+    FOREIGN KEY (application_uuid)
+    REFERENCES application (uuid),
     CONSTRAINT fk_offer_connection_uuid
     FOREIGN KEY (offer_connection_uuid)
     REFERENCES offer_connection (uuid)
 );
 
 -- application_remote_relation represents a look up table to find the consumer
--- relation UUID for a given local relation.
+-- relation UUID for a given (syntethic) relation in the offerer model.
 CREATE TABLE application_remote_relation (
-    uuid TEXT NOT NULL PRIMARY KEY,
-    -- relation_uuid is the local relation UUID.
-    relation_uuid TEXT NOT NULL,
+    -- relation_uuid is the relation UUID as created in the offerer model. This
+    -- is effectively a synthetic relation.
+    relation_uuid TEXT NOT NULL PRIMARY KEY,
     -- consumer_relation_uuid is the relation UUID in the consumer model.
     -- There is no FK constraint on it, because we don't have the relation
     -- locally in the model.
+    -- NOTE: In terms of the cross model relation API, this is the 
+    -- relation-token.
     consumer_relation_uuid TEXT NOT NULL,
     CONSTRAINT fk_relation_uuid
     FOREIGN KEY (relation_uuid)
     REFERENCES relation (uuid)
 );
 
+-- Offering model relations (synthetic) are 1:1 with consumer model relations.
+CREATE UNIQUE INDEX idx_application_remote_relation_consumer_relation_uuid
+ON application_remote_relation (consumer_relation_uuid);
```

Comment:
Maybe just setting `consumer_relation_uuid` to be the primary key is better?

Reply:
No because in that case you could have 1:N to relationUUID (the synthetic ones). 
Even doing `PRIMARY KEY (relation_uuid, consumer_relation_uuid)` wouldn't be good enough, since what we want is to only have 1:1 between local and remote relations.